### PR TITLE
rclcpp::shutdown should not be called before LifecycleNode dtor.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address part of https://github.com/ros2/rclcpp/issues/2520

### CODE REVIEW

**Comment by fujitatomoya** [comment]
@clalancette can you take a look?

**Comment by fujitatomoya** [comment]
`LifecyclePublisher` is available in `humble`, so we need to backport this to all distro.

File: rclcpp_lifecycle/test/test_lifecycle_publisher.cpp [hunk 1]
```diff
@@ -93,95 +79,103 @@ class TestLifecyclePublisher : public ::testing::TestWithParam<TimerType>
   void SetUp()
   {
     rclcpp::init(0, nullptr);
-    node_ = std::make_shared<EmptyLifecycleNode>("node", GetParam());
```
[the remark this answers is not in the document]
Comment (fujitatomoya) [reply rclcpp_lifecycle/test/test_lifecycle_publisher.cpp:96]: thanks, that is what we can do. i do not have preference, just a single line code to create node for each test is not so bad i think.

**Comment by fujitatomoya** [comment]
@mjcarroll @alsora i could use review for this. after i have a lgtm, i will start the CI.

**Review by mjcarroll — APPROVED** [review]
LGTM with CI

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21140/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15441/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21863/)

**Comment by fujitatomoya** [comment]
aarch64 test failures are unrelated, https://ci.ros2.org/job/ci_linux-aarch64/15441/

**Comment by fujitatomoya** [comment]
@Mergifyio backport humble iron jazzy

**Pull request merged**